biodatasets/meddocan/meddocan.py

```diff
@@ -0,0 +1,273 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+A dataset loading script for the MEDDOCAN corpus.
+The MEDDOCAN datset is a manually annotated collection of clinical case
+reports derived from the Spanish Clinical Case Corpus (SPACCC). It was designed
+for the Medical Document Anonymization Track, the first the first community
+challenge task specifically devoted to the anonymization of medical documents in Spanish
+"""
+
+import os
+import datasets
+from pathlib import Path
+from typing import List, Tuple, Dict
+from utils import parsing, schemas
+from utils.configs import BigBioConfig
+from utils.constants import Tasks
+
+_CITATION = """\
+@inproceedings{marimon2019automatic,
+  title={Automatic De-identification of Medical Texts in Spanish: the MEDDOCAN Track, Corpus, Guidelines, Methods and Evaluation of Results.},
+  author={Marimon, Montserrat and Gonzalez-Agirre, Aitor and Intxaurrondo, Ander and Rodriguez, Heidy and Martin, Jose Lopez and Villegas, Marta and Krallinger, Martin},
+  booktitle={IberLEF@ SEPLN},
+  pages={618--638},
+  year={2019}
+}
+"""
+
+_DATASETNAME = "meddocan"
+
+_DESCRIPTION = """\
+MEDDOCAN: Medical Document Anonymization Track
+
+This dataset is designed for the MEDDOCAN task, sponsored by Plan de Impulso de las Tecnologías del Lenguaje.
+
+It is a manually classified collection of 1,000 clinical case reports derived from the \
+Spanish Clinical Case Corpus (SPACCC), enriched with PHI expressions.
+
+The annotation of the entire set of entity mentions was carried out by experts annotators\
+and it includes 29 entity types relevant for the annonymiation of medical documents.\
+22 of these annotation types are actually present in the corpus: TERRITORIO, FECHAS, \
+EDAD_SUJETO_ASISTENCIA, NOMBRE_SUJETO_ASISTENCIA, NOMBRE_PERSONAL_SANITARIO, \
+SEXO_SUJETO_ASISTENCIA, CALLE, PAIS, ID_SUJETO_ASISTENCIA, CORREO, ID_TITULACION_PERSONAL_SANITARIO,\
+ID_ASEGURAMIENTO, HOSPITAL, FAMILIARES_SUJETO_ASISTENCIA, INSTITUCION, ID_CONTACTO ASISTENCIAL,\
+NUMERO_TELEFONO, PROFESION, NUMERO_FAX, OTROS_SUJETO_ASISTENCIA, CENTRO_SALUD, ID_EMPLEO_PERSONAL_SANITARIO
+
+The PharmaCoNER corpus contains around 495 thousand words with an average of 494 words per clinical case.
+The 1,000 clinical case reports have been randomly sampled into 3 subsets. 
+The training set contains 500 clinical cases, while the development \
+and test sets contain 250 clinical cases each.
+    
+For further information, please visit https://temu.bsc.es/meddocan/ or send an email to encargo-pln-life@bsc.es
+"""
+
+
+_HOMEPAGE = "https://temu.bsc.es/meddocan/index.php/datasets/"
```

Comment:
I would use this as the official webpage "https://temu.bsc.es/meddocan/", it;s got a better description of the dataset and task